### PR TITLE
Issue/316 validate pf

### PR DESCRIPTION
**Pull request by AntoineRondelet**
see #316

### CODE REVIEW

File: ttnctl/cmd/applications_pf_set.go [hunk 1]
```diff
@@ -140,7 +155,58 @@ function Decoder(bytes, port) {
...
+			default:
+				ctx.Fatalf("Function %s does not exist", function)
+			}
+		}
+
+		fmt.Printf("\nDo you want to test the payload function ? (Y/n)\n")
+		var response string
+		fmt.Scanln(&response)
+
+		if strings.ToLower(response) == "y" || strings.ToLower(response) == "yes" {
```
Comment (htdvisser) [inline ttnctl/cmd/applications_pf_set.go]: Add `""` (Y means default yes)

File: ttnctl/cmd/applications_pf_set.go [hunk 2]
```diff
@@ -140,7 +155,58 @@ function Decoder(bytes, port) {
...
+			switch function {
+			case "decoder":
```
Comment (htdvisser) [inline ttnctl/cmd/applications_pf_set.go]: `case "decoder", "converter", "validator":`

File: ttnctl/util/cli_input.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	fmt.Print("Port: ")
+	input, err := ReadInput(stdin)
+	if err != nil {
+		return 0, err
+	}
+
+	port, err := strconv.ParseUint(input, 10, 8)
+	if err != nil {
+		if IsErrOutOfRange(err) {
+			return 0, fmt.Errorf(fmt.Sprintf("The port number is too big (Should be int8): %s", err.Error()))
```
Comment (htdvisser) [inline ttnctl/util/cli_input.go]: `uint8`

File: ttnctl/util/cli_input.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func ReadPayload() ([]byte, error) {
+	fmt.Print("Payload: ")
+	input, err := ReadInput(stdin)
+	if err != nil {
+		return nil, err
+	}
+
+	payload := []byte(input)
```
Comment (htdvisser) [inline ttnctl/util/cli_input.go]: Remove spaces and decode hex

File: ttnctl/cmd/applications_pf_set.go [hunk 5]
```diff
@@ -140,7 +155,54 @@ function Decoder(bytes, port) {
...
 }
 ########## Write your Encoder here and end with Ctrl+D (EOF):`)
-				app.Encoder = readFunction()
+				app.Encoder = readFunction(ctx)
+			default:
+				ctx.Fatalf("Function %s does not exist", function)
+			}
+		}
+
+		fmt.Printf("\nDo you want to test the payload function ? (Y/n)\n")
```
Comment (romeovs) [inline ttnctl/cmd/applications_pf_set.go]: `functions`

Comment (romeovs) [inline ttnctl/cmd/applications_pf_set.go]: Also, remove the space before `?`

File: ttnctl/cmd/applications_pf_set.go [hunk 6]
```diff
@@ -44,6 +44,15 @@ function Decoder(bytes, port) {
 
   return decoded;
 }
+
+Do you want to test the payload function ? (Y/n)
```
Comment (romeovs) [inline ttnctl/cmd/applications_pf_set.go]: `functions?`

**Pull request merged**